wnfs/src/private/file.rs

```diff
@@ -342,6 +342,51 @@ impl PrivateFile {
         })
     }
 
+    /// Reads a number of bytes starting from a given offset.
+    pub async fn read_at<'a>(
+        &'a self,
+        offset: usize,
+        size: usize,
+        forest: &'a PrivateForest,
+        store: &'a impl BlockStore,
+    ) -> Result<Vec<u8>> {
+        let block_content_size = MAX_BLOCK_CONTENT_SIZE;
+        let chunk_size_upper_bound = (self.get_content_size_upper_bound() - offset).min(size);
+        if chunk_size_upper_bound == 0 {
+            return Ok(vec![]);
+        }
+        let first_block = offset / block_content_size;
+        let last_block = (offset + size) / block_content_size;
+        let mut bytes = Vec::with_capacity(chunk_size_upper_bound);
+        let mut content_stream = self.stream_content(first_block, forest, store).enumerate();
+        while let Some((i, chunk)) = content_stream.next().await {
+            match chunk {
+                Ok(chunk) => {
+                    let index = first_block + i;
+                    let from = if index == first_block {
+                        (offset - index * block_content_size).min(chunk.len())
+                    } else {
+                        0
+                    };
+                    let to = if index == last_block {
+                        (offset + size - index * block_content_size).min(chunk.len())
+                    } else {
+                        chunk.len()
+                    };
+                    bytes.extend_from_slice(&chunk[from..to]);
+                    if index == last_block {
+                        break;
+                    }
+                }
+                // If a block doesn't exist, break and return what we read until here.
+                Err(_err) => {
+                    break;
+                }
```

Comment:
Can you make this just error out?
We shouldn't return any bytes if the content is corrupted. Usually we'd do `InvalidSerialization` or w/e it was called, but adding a custom error type may not be bad either.

Reply:
Yup, makes sense.